src/test/java/edu/ksu/canvas/attendance/submitter/AssignmentSubmitterUTest.java

```diff
@@ -0,0 +1,240 @@
+        Whitebox.setInternalState(assignmentSubmitter, "assignmentValidator", assignmentValidator);
+
+
+
+    }
+
+    @Test
+    public void submitCourseAttendancesHappyPath() throws IOException, CanvasOutOfSyncException {
+        when(sectionService.getSectionInListById(COURSE_ID, SECTION_1_ID)).thenReturn(section1);
+        when(sectionService.getSectionInListById(COURSE_ID, SECTION_2_ID)).thenReturn(section2);
+        when(assignmentService.findBySection(section1)).thenReturn(attendanceAssignment1);
+        when(assignmentService.findBySection(section2)).thenReturn(attendanceAssignment2);
+        when(canvasApiWrapperService.getSingleAssignment(COURSE_ID, oauthToken, CANVAS_ASSIGNMENT_ID + "")).thenReturn(assignmentOptional);
+        when(attendanceCourseService.findByCanvasCourseId(eq(COURSE_ID))).thenReturn(course);
+        when(attendanceService.getAttendanceCommentsBySectionId(SECTION_1_ID)).thenReturn(studentCommentsMap1);
+        when(attendanceService.getAttendanceCommentsBySectionId(SECTION_2_ID)).thenReturn(studentCommentsMap2);
+        when(canvasApiWrapperService.gradeMultipleSubmissionsBySection(any(), any())).thenReturn(progressOptional);
+
+        assignmentSubmitter.submitCourseAttendances(true, attendanceSummaryModelList, COURSE_ID, oauthToken, assignmentConfigurationFromSetup);
+        verify(canvasApiWrapperService, times(2)).gradeMultipleSubmissionsBySection(any(), any());
+    }
+
+    @Test
+    public void submitCourseAttendancesFailCanvasGradingError() throws IOException, CanvasOutOfSyncException {
+        progressOptional.get().setWorkflowState("failed");
+
+        when(sectionService.getSectionInListById(COURSE_ID, SECTION_1_ID)).thenReturn(section1);
+        when(sectionService.getSectionInListById(COURSE_ID, SECTION_2_ID)).thenReturn(section2);
+        when(assignmentService.findBySection(section1)).thenReturn(attendanceAssignment1);
+        when(assignmentService.findBySection(section2)).thenReturn(attendanceAssignment2);
+        when(canvasApiWrapperService.getSingleAssignment(COURSE_ID, oauthToken, CANVAS_ASSIGNMENT_ID + "")).thenReturn(assignmentOptional);
+        when(attendanceCourseService.findByCanvasCourseId(eq(COURSE_ID))).thenReturn(course);
+        when(attendanceService.getAttendanceCommentsBySectionId(SECTION_1_ID)).thenReturn(studentCommentsMap1);
+        when(attendanceService.getAttendanceCommentsBySectionId(SECTION_2_ID)).thenReturn(studentCommentsMap2);
+        when(canvasApiWrapperService.gradeMultipleSubmissionsBySection(any(), any())).thenReturn(progressOptional);
+
+        try {
+            assignmentSubmitter.submitCourseAttendances(true, attendanceSummaryModelList, COURSE_ID, oauthToken, assignmentConfigurationFromSetup);
+            verify(canvasApiWrapperService, times(1)).gradeMultipleSubmissionsBySection(any(), any());
+        } catch (IOException exception) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1166)